crates/holochain/src/conductor/conductor/conductor_services/dpki_service.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Arc;
+
+use holo_hash::AgentPubKey;
+use holochain_zome_types::{CellId, Timestamp};
+
+use crate::conductor::ConductorHandle;
+
+/// Interface for the DPKI service
+#[async_trait::async_trait]
+#[mockall::automock]
+#[allow(clippy::needless_lifetimes)]
+pub trait DpkiService: Send + Sync {
+    /// Check if the key is valid (properly created and not revoked) as-at the given Timestamp
+    async fn is_key_valid(&self, key: AgentPubKey, timestamp: Timestamp)
+        -> DpkiServiceResult<bool>;
+
+    /// Defines the different ways that keys can be created and destroyed:
+    /// If an old_key is specified, it will be destroyed
+    /// If a new key is specified, it will be registered
+    /// If both a new and an old key are specified, the new key will atomically replace the old key
+    /// (If no keys are specified, nothing will happen)
```

Comment:
output a warning?

Reply:
That would be the responsibility of the implementor, and there is no concrete implementation for this method yet.

Reply:
But it could be part of the implementation when the time comes.